python/pyfunc-server/examples/echo_upi/upi_server.py

```diff
@@ -30,12 +30,23 @@ def upiv1_infer(self, request: upi_pb2.PredictValuesRequest,
             metadata=upi_pb2.ResponseMetadata(
                 prediction_id=request.metadata.prediction_id,
                 # TODO: allow user to get model name and version from PyFuncModel
-                models=[upi_pb2.ModelMetadata(name=self._model_name, version=self._model_version)]
-            )
+                models=[
+                    upi_pb2.ModelMetadata(
+                        name=self._model_name, version=self._model_version
+                    )
+                ],
+            ),
         )
 
 
 if __name__ == "__main__":
     model_name = "echo-model"
     model_version = "1"
-    mlflow.pyfunc.log_model("model", python_model=EchoUPIModel(model_name, model_version))
+
+    merlin.run_pyfunc_model(
```

Comment:
In order to start local GRPC server, we specify `env_vars` rather than use `Protocol ` in the argument?

Reply:
Yes, you're correct. When deploying from UI or SDK, they can set the protocol argument (e.g. [`protocol` in `deploy()` function.](https://github.com/caraml-dev/merlin/blob/main/python/sdk/merlin/fluent.py#L359)). Then the API will set the environment variables for them ([ref](https://github.com/caraml-dev/merlin/blob/main/api/cluster/resource/templater.go#L710)). To have similar behaviour as this, I let the user to set the same environment variables here. But I think we can have `protocol` argument in `run_pyfunc_model` too. Wdyt? cc: @tiopramayudi

Reply:
yup that would be better, update env vars manually by user is prone to error

Reply:
you may mark this as resolve (I can't)